rs/sns/governance/proto/ic_sns_governance/pb/v1/governance.proto

```diff
@@ -2138,6 +2139,46 @@ message AdvanceTargetVersionRequest {
 // The response to a request to advance the target version of the SNS.
 message AdvanceTargetVersionResponse {}
 
+// Represents a single entry in the upgrade journal.
+message UpgradeJournalEntry {
+  oneof entry {
+    UpgradeStepsRefresh upgrade_steps_refresh = 1;
+    TargetVersionSet target_version_set = 2;
+    UpgradeStarted upgrade_started = 3;
+    UpgradeCompleted upgrade_completed = 4;
+  }
+  optional uint64 timestamp_seconds = 5;
+
+  message UpgradeStepsRefresh {
+    optional Governance.Versions upgrade_steps = 2;
+  }
+
+  message TargetVersionSet {
+    optional Governance.Version version = 2;
+  }
+
+  message UpgradeStarted {
+    optional Governance.Version current_version = 2;
+    optional Governance.Version expected_version = 3;
+  }
+
+  message UpgradeCompleted {
+    oneof status {
+      Empty succeeded = 1;
+      Empty timed_out = 2;
+      Governance.Version invalid_state = 3;
+      Empty canister_call_failure = 4;
+    }
```

Comment:
@aterga This wasn't part of the original design we discussed, but I think it may make more sense to have `UpgradeCompleted` be a separate event from `UpgradeStarted` - this means we can always just append to the journal instead of having to go in and edit previous entries, which simplifies the implementation and better fits the actual implementation IMO